src/Runner.Worker/ContainerOperationProvider.cs

```diff
@@ -299,15 +327,20 @@ private async Task StopContainerAsync(IExecutionContext executionContext, Contai
 
             if (!string.IsNullOrEmpty(container.ContainerId))
             {
-                if (!container.IsJobContainer)
+                if (!container.IsJobContainer && !container.FailedInitialization)
                 {
-                    // Print logs for service container jobs (not the "action" job itself b/c that's already logged).
-                    executionContext.Output($"Print service container logs: {container.ContainerDisplayName}");
-
-                    int logsExitCode = await _dockerManager.DockerLogs(executionContext, container.ContainerId);
-                    if (logsExitCode != 0)
+                    var healthcheck = await Healthcheck(executionContext, container);
+                    if (string.Equals(healthcheck, "healthy", StringComparison.OrdinalIgnoreCase))
```

Comment:
what about the case where `healthcheck==""`? do we want to print the log as well?

Reply:
If there is no healthcheck or it is empty we assume that the container is healthy and don't check the logs in the 'Initialize Containers' section. They will be printed in the 'Stop containers' section. I am open to suggestions here - we can also treat containers with unknown status as unhealthy.

Reply:
the current code is in `StopContainerAsync`, will it print out the log from any service container without health check result?

Reply:
Sorry for the confusion - if the healthcheck is anything else than "healthy" it is considered unhealthy and printed in the 'start containers' section.

Reply:
Will that fail the job? is this changing the existing behavior?

Reply:
Yes, if there is no healthcheck we will fail the job and print it's logs in 'Stop containers' section.